OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs

```diff
@@ -15,13 +15,25 @@ namespace OpenRA.Mods.Common.Widgets
 {
 	public static class ConfirmationDialogs
 	{
-		public static void PromptConfirmAction(string title, string text, Action onConfirm, Action onCancel = null, string confirmText = null, string cancelText = null)
+		public static void PromptConfirmAction(
+			string title,
+			string text,
+			Action onConfirm,
+			Action onCancel = null,
+			Action onRestart = null,
+			string confirmText = null,
+			string cancelText = null,
+			string restartText = null)
 		{
 			var prompt = Ui.OpenWindow("CONFIRM_PROMPT");
+			var restartButton = prompt.GetOrNull<ButtonWidget>("RESTART_BUTTON");
```

Comment:
Could you pull out variables for `confirmButton` and `cancelButton` while you're here too?  That saves two unnecessary widget lookups.